python/tank_vendor/shotgun_authentication/console_authentication.py

```diff
@@ -101,20 +119,17 @@ def _get_keyboard_input(self, label, default_value=""):
             user_input = raw_input(text) or default_value
         return user_input
 
-    def _get_session_token(self, hostname, login, password, http_proxy):
+    def _get_2fa_code(self):
         """
-        Retrieves a session token for the given credentials. If it fails, the user is informed
-        :param hostname: The host to connect to.
-        :param login: The user to get a session for.
-        :param password: Password for the user.
-        :param http_proxy: Proxy to use. Can be None.
-        :returns: If the credentials were valid, returns a session token, otherwise returns None.
+        Prompts the user for his 2fa code.
+        :returns: Two factor authentication code.
+        :raises AuthenticationCancelled: If the user enters an empty code, the exception will be
+                                         thrown.
         """
-        try:
-            return session_cache.generate_session_token(hostname, login, password, http_proxy)
-        except AuthenticationError:
-            print "Login failed."
-            return None
+        code = raw_input("Code (empty to abort): ")
```

Comment:
This feels like a bit too blocky. It's harder to know what's going on and where you are supposed to look at.
![screen shot 2015-06-09 at 10 44 43 am](https://cloud.githubusercontent.com/assets/8126447/8060701/c31526a4-0e94-11e5-90a3-713139a47361.png)
This makes it unnecessarily long:
![screen shot 2015-06-09 at 10 44 32 am](https://cloud.githubusercontent.com/assets/8126447/8060723/e59054ec-0e94-11e5-9b0b-9eebee00a2c3.png)
I feel like a compromise between the two is the best:
![screen shot 2015-06-09 at 10 36 25 am](https://cloud.githubusercontent.com/assets/8126447/8060735/f7f071da-0e94-11e5-822a-93d396990203.png)
What do you think?